`textlint --dry-run` exits with non-zero code

## Description

Related: #1461

[Docs](https://textlint.github.io/docs/cli.html#exit-code) says: 

> 0: No Error
> --dryRun: Found lint error but --dryRun is specified

But `textlint` returns non-zero exit code even if I specified `--dry-run`. Is it intended?

## Log

```bash
mkdir textlint
cd textlint

cat <<'EOF'>README.md
# test

TODO: this is test
EOF

npm install --save-dev textlint textlint-rule-no-todo
```

```shellsession
$ cat package.json
{
  "devDependencies": {
    "textlint": "^14.4.2",
    "textlint-rule-no-todo": "^2.0.1"
  }
}
$ npx textlint -v
v14.4.2
$ npx textlint ./README.md --dry-run --rule no-todo

/home/eggplants/prog/textlint-test/README.md
  3:1  error  Found TODO: 'TODO: this is test'  no-todo

✖ 1 problem (1 error, 0 warnings)

$ echo $?
1
```

## Environment

```shellsession
$ uname -v
#19~24.04.1-Ubuntu SMP PREEMPT_DYNAMIC Mon Feb 17 11:51:52 UTC 2
$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 24.04.2 LTS
Release:        24.04
Codename:       noble
$ node -v
v20.12.2
```

It seems that `textlint --fix` always returns `0` (except with unexpected error?) because non-fixable rules seem to be ignored when either `--fix` or `--fix --dry-run` is given.

<details>

```
mkdir textlint
cd textlint

cat <<'EOF'>README.md
# test.

TODO: this is test.
EOF

npm install --save-dev textlint textlint-rule-no-todo textlint-rule-period-in-header
```

```shellsession
$ npx textlint ./README.md  --rule textlint-rule-period-in-header --rule no-todo

/home/eggplants/prog/textlint-test/README.md
  1:7  ✓ error  Should remove period mark(".") at end of header  period-in-header
  3:1  error    Found TODO: 'TODO: this is test.'                no-todo

✖ 2 problems (2 errors, 0 warnings)
✓ 1 fixable problem.
Try to run: $ textlint --fix [file]

$ echo $?
1
$ npx textlint ./README.md  --rule textlint-rule-period-in-header --rule no-todo --fix --dry-run

/home/eggplants/prog/textlint-test/README.md
  1:7  ✔   Should remove period mark(".") at end of header  period-in-header

✔ Fixed 1 problem

$ echo $?
0
$ npx textlint ./README.md  --rule textlint-rule-period-in-header --rule no-todo --fix

/home/eggplants/prog/textlint-test/README.md
  1:7  ✔   Should remove period mark(".") at end of header  period-in-header

✔ Fixed 1 problem

$ echo $?
0
$ npx textlint ./README.md  --rule textlint-rule-period-in-header --rule no-todo --fix
$ echo $?
0
$ npx textlint ./README.md  --rule textlint-rule-period-in-header --rule no-todo

/home/eggplants/prog/textlint-test/README.md
  3:1  error  Found TODO: 'TODO: this is test.'  no-todo

✖ 1 problem (1 error, 0 warnings)

$ echo $?
1
```

</details>

> I feel the documentation is wrong.

Am I correct that we should remove the statement `--dryRun: Found lint error but --dryRun is specified` from documentation?

--- a/docs/cli.md
+++ b/docs/cli.md
@@ -117,12 +117,12 @@ If you want to clear the cache, you can use the `--no-cache` option or just remo
 - Not found lint error
 - --fix: found errors but fix all errors, so exit with 0
 - --output-file: Found lint error but --output-file is specified
-- --dry-run: Found lint error but --dry-run is specified
 
 `1`: Lint Error
 
 - found lint error
 - --fix: found errors and could not fix all errors, so exit with 1
+- --dry-run: Found lint error but --dry-run is specified hence not fixed
 
 `2`: Fatal Error
 

